core/domain/exp_domain.py

```diff
@@ -2158,10 +2160,13 @@ def _convert_states_v24_dict_to_v25_dict(cls, exp_id, states_dict):
         Returns:
             dict. The converted states_dict.
         """
+        file_system_class = fs_services.get_entity_file_system_class()
+        file_system = fs_domain.AbstractFileSystem(file_system_class(
```

Comment:
In this file (in multiple places) and in previous files, maybe call this variable entity_file_system, to make it clear that it is not the general filesystem but the one specifically related to a particular entity.